src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/TargetSignatureTests.cs

```diff
@@ -194,6 +194,53 @@ partial interface IComInterface
                 }
                 """;
 
+            await VerifyComInterfaceGeneratorAsync(source, "IComInterface", "Method", (newComp, signature) =>
+            {
+                Assert.Equal(SignatureCallingConvention.Unmanaged, signature.CallingConvention);
+                Assert.Equal(newComp.GetTypeByMetadataName("System.Runtime.CompilerServices.CallConvMemberFunction"), Assert.Single(signature.UnmanagedCallingConventionTypes), SymbolEqualityComparer.Default);
+            });
+        }
+
+        [Fact]
+        public async Task ComInterfacePreserveSigMethodHasMemberFunctionCallingConventionByDefault()
+        {
+            string source = $$"""
+                using System.Runtime.CompilerServices;
+                using System.Runtime.InteropServices;
+                using System.Runtime.InteropServices.Marshalling;
+
+                [GeneratedComInterface]
+                [Guid("0A617667-4961-4F90-B74F-6DC368E98179")]
+                partial interface IComInterface
+                {
+                    [PreserveSig]
+                    int Method();
```

Comment:
Wasn't there another test with this, can we validate there instead of here?

Reply:
I'm thinking of the implementing PR.

Reply:
I'd rather have a separate test for the calling convention vs the signature types. I like having fine-grained unit tests that each assert one concept if possible.

Reply:
I would reconsider this philosophy. It quickly scales out of control and leads to redirection and unnatural acts as we copy identical code around. Tests that contain unique input should be fully validated rather than having another test run that does the precise same work. The assert itself tells you the issue not the input to the test case. If our infrastructure is limited to a single assert/concept to validate then it is broken.